### PR TITLE
chore(workflows): split release workflow into two for SOC 2 compliance

### PR DESCRIPTION
**Pull request by electrosenpai**
This Pull Request separates the release workflow into two distinct workflows to comply with SOC 2 requirements : we can't directly commit on main even if its a bot.

- Workflow 1: Creates a release Pull Request to update version numbers.
- Workflow 2: Publishes the Helm charts upon merging the release PR into the main branch.

This ensures that all code changes are reviewed and approved before being merged, aligning with our compliance standards.

### CODE REVIEW

File: .github/workflows/creating_pr.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Create Release PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to release'
+        required: true
+        default: '1.0.0'
```
Comment (jdenquin) [inline .github/workflows/creating_pr.yml:9]: it sounds like we may have `1.0.0`  often, can we remove the default?

File: .github/workflows/creating_pr.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Commit changes
+        run: |
+          git checkout -b release/${{ github.event.inputs.tag }}
+          git add Chart.yaml values.yaml
+          git commit -m "chore(release): update chart, appVersion, and values versions to ${{ github.event.inputs.tag }}"
+          git push origin release/${{ github.event.inputs.tag }}
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v5
```
Comment (jdenquin) [inline .github/workflows/creating_pr.yml:37]: can we use the last version? `v7`  ?

Comment (electrosenpai) [reply .github/workflows/creating_pr.yml:37]: good catch here thanks !

File: .github/workflows/creating_pr.yml (hunk 1 above)
Comment (electrosenpai) [reply .github/workflows/creating_pr.yml:9]: yes sure !

File: .github/workflows/creating_pr.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        run: |
+          git checkout -b release/${{ github.event.inputs.tag }}
+          git add Chart.yaml values.yaml
+          git commit -m "chore(release): update chart, appVersion, and values versions to ${{ github.event.inputs.tag }}"
+          git push origin release/${{ github.event.inputs.tag }}
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v7
+        with:
+          commit-message: "chore(release): update chart, appVersion, and values versions to ${{ github.event.inputs.tag }}"
```
Comment (jdenquin) [inline .github/workflows/creating_pr.yml:38]: `update to`  is enough?

Comment (jdenquin) [reply .github/workflows/creating_pr.yml:38]: yes sir!

**Review by jdenquin — APPROVED** [review]

**Pull request merged**